src/pages/components/Overlay.qml

```diff
@@ -415,15 +419,23 @@ PanelBackground {
 
             onStatusChanged: {
                 if (activeWebPage && status == PageStatus.Active) {
-                    activeWebPage.grabToFile()
+                    webView.privateMode ? activeWebPage.grabThumbnail() : activeWebPage.grabToFile()
                 }
             }
 
             Browser.TabView {
                 model: webView.tabModel
                 portrait: tabPage.isPortrait
+                privateMode: webView.privateMode
 
                 onHide: pageStack.pop()
+
+                onPrivateModeChanged: {
+                    webView.privateMode = privateMode
+                    tabPage.activeTabIndex =  webView.tabModel.activeTabIndex
+                    tabPage.activeWebPage = webView.contentItem
```

Comment:
With above code consider following sequence of events :

1) Normal browsing active
2) Restart browser
3) Switch to private mode
4) Switch back to normal mode
4) Page pop the page

the box to enter url is visible even when tabmodel in normal browsing mode is not empty.

I propose below code :
```
                    if (webView.tabModel.count === 0) {
                        overlay.enterNewTabUrl(PageStackAction.Immediate)
                    } else {
                        if (overlayAnimator.state !== "fullscreenWebPage") {
                            dismiss()
                        }
                    }
```